torchtune/modules/peft/dora.py

```diff
@@ -104,11 +104,28 @@ def initialize_dora_magnitude(self):
         """
         DoRA initializes the magnitude vector such that its outputs are initially
         identical to standard LoRA's outputs.
+
+        This must be called after loading/initializing base model and LoRA params.
         """
+        if any(
+            [
+                self.weight.is_meta,
+                self.lora_a.weight.is_meta,
+                self.lora_b.weight.is_meta,
+            ]
+        ):
+            raise ValueError(
+                "Cannot initialize DoRA magnitude until after base and LoRA parameters"
+            )
         base_weight = self.weight.to(self.lora_a.weight.dtype)
         lora_weight = self.lora_b.weight @ self.lora_a.weight
-        weight_norm = self._get_weight_norm(base_weight, lora_weight)
-        self.magnitude.copy_(weight_norm)
+
+        magnitude = nn.Parameter(
+            torch.empty_like(self.magnitude, device=self.lora_a.weight.device),
+            requires_grad=self.magnitude.requires_grad,
```

Comment:
When would this be `False`?

Reply:
Offhand I can’t think of a case but imo better to just be explicit here